rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "1.71.0"
```

Comment:
Does this mean that a user with a default toolchain version of 1.72.0 (once that's out) will still run the build with 1.71.0? It seemed that way when I tried it just now by adding a `rust-toolchain.toml` file with `channel = "1.60.0"`. It failed with this:
```
error: package `toml_datetime v0.6.3` cannot be built because it requires rustc 1.64.0 or newer, while the currently active rustc version is 1.60.0
```

Reply:
Yeah, if you're using `rustup`, it should automatically download and setup the correct version of the Rust compiler. Older versions won't work (as in your example), but newer versions should be fine. They'll just use that specific version of `rustc` when working in the repo.

Reply:
What I mean is that this seems to use version to 1.71 even if the user's default toolchain is 1.72. Is that what we want?

Reply:
Update: we discussed this on Discord, and I think the answer is "yes, this is what we want." Basically, we're going to invert the situation now where people use whatever Rust toolchain they have, and CI tests MSRV, and go in reverse: everyone develops with the MSRV, and the CI checks every change against the latest stable. The intent is to mainly catch diagnostics, clippy warnings, etc.

Reply:
I've also split out the `rust-toolchain.toml` change anyway, so we'll handle it in an upcoming idff.